cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2673 @@
+       "      <td>True</td>\n",
+       "      <td>0.989083</td>\n",
+       "      <td>&lt;NA&gt;</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>255</th>\n",
+       "      <td>ÐºÐ°Ð¼Ð¿Ð°Ð½Ð¸Ð¸ Ð¸ Ñ€ÐµÐºÐ»Ð°Ð¼Ð½Ñ‹Ðµ Ð°ÐºÑ†Ð...</td>\n",
+       "      <td>True</td>\n",
+       "      <td>0.988594</td>\n",
+       "      <td>&lt;NA&gt;</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2071</th>\n",
+       "      <td>Ð²Ð°ÑˆÐ¸Ñ… Ñ†ÐµÐ»ÐµÐ¹ Ð² Ñ​ÐµÐ»ÑŒÑ​ÐºÐ¾Ð¼ Ñ…Ð¾...</td>\n",
+       "      <td>True</td>\n",
+       "      <td>0.987874</td>\n",
+       "      <td>&lt;NA&gt;</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>515</th>\n",
+       "      <td>Ñ€Ð°Ð·Ñ€Ð°Ð±Ð¾Ñ‚ÐºÐµ ÑƒÐ¼Ð½Ñ‹Ñ… Ñ​ÐµÐ»ÑŒÑ​ÐºÐ¾...</td>\n",
+       "      <td>True</td>\n",
+       "      <td>0.987554</td>\n",
+       "      <td>&lt;NA&gt;</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                   text  is_non_english  \\\n",
+       "127   Ð²Ñ‹Ñ​Ñ‚Ð°Ð²ÐºÐ°Ñ… Ð¸ ÐºÐ¾Ð½Ñ„ÐµÑ€ÐµÐ½Ñ†Ð¸Ñ​Ñ…...            True   \n",
+       "256   Ð¾Ð·Ñ​Ð¹Ñ​Ñ‚Ð²ÐµÐ½Ð½Ñ‹Ñ… Ñ€ÐµÑˆÐµÐ½Ð¸Ð¹ Ð´Ð»Ñ​...            True   \n",
+       "255   ÐºÐ°Ð¼Ð¿Ð°Ð½Ð¸Ð¸ Ð¸ Ñ€ÐµÐºÐ»Ð°Ð¼Ð½Ñ‹Ðµ Ð°ÐºÑ†Ð...            True   \n",
+       "2071  Ð²Ð°ÑˆÐ¸Ñ… Ñ†ÐµÐ»ÐµÐ¹ Ð² Ñ​ÐµÐ»ÑŒÑ​ÐºÐ¾Ð¼ Ñ…Ð¾...            True   \n",
+       "515   Ñ€Ð°Ð·Ñ€Ð°Ð±Ð¾Ñ‚ÐºÐµ ÑƒÐ¼Ð½Ñ‹Ñ… Ñ​ÐµÐ»ÑŒÑ​ÐºÐ¾...            True   \n",
+       "\n",
+       "      non_english_score predicted_language  \n",
+       "127            0.989548               <NA>  \n",
+       "256            0.989083               <NA>  \n",
+       "255            0.988594               <NA>  \n",
+       "2071           0.987874               <NA>  \n",
+       "515            0.987554               <NA>  "
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "non_english_issues = cleanlab_df.query(\"is_non_english\").sort_values(\"non_english_score\", ascending=False)\n",
+    "\n",
+    "non_english_columns = ['is_non_english', 'non_english_score', 'predicted_language']\n",
+    "columns_to_display = [text_column] + non_english_columns\n",
+    "\n",
+    "print(f\"There are {len(non_english_issues)} non English issues in our documents data. Here are some examples below:\\n\")\n",
+    "display(non_english_issues.head(5)[columns_to_display])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d288882",
+   "metadata": {},
+   "source": [
+    "We have observed the capabilities that Cleanlab Studio can provide in detecting these text-specific issues, so let's now:\n",
+    "\n",
+    "1. Filter for the chunks that have the issues detected by Cleanlab\n",
+    "2. Update our Pinecone DB by removing the chunks with any of the issues detected by Cleanlab\n",
```

Comment:
This is above my paygrade but IMO it'd be better to tag with metadata in pinecone than to delete. That way the RAG system can say "hey, for these queries, ignore toxic content but not informal, etc" via metadata filter

I think this makes our demo more attractive to pinecone too because pinecone is great at metadata filtering and it's more interactive Pinecone<>Cleanlab  as opposed to just a pre-cleaning step.